docs/aep/AEP-2-table-cleanup.md

```diff
@@ -0,0 +1,186 @@
+            conn_id="postgres_conn",
+        ),
+    )
+    sample_table = sample_create_table(my_homes_table)
+    my_df_func(sample_table)
+```
+
+Pros:
+* Agnostic to DAG shape
+* More transparent than DAG object
+
+Cons:
+* Users can no longer supply their own on_success or on_failure callbacks
+* Requires some airflow knowledge (knowing that these callbacks exist)
+* Callback logs are not easily viewable to users
+
+### Option 3: a function with a table parameter
+
+If we want FULL user transparency, we could offer a “cleanup” function that creates a task. It’s worth noting that as a task, there’s no implicit way to place this function at the end of the DAG, therefore we would ask users supply the tasks as function parameters s.t. We can clean the tasks at the end
+
+```python
+with dag:
+    my_homes_table = aql.load_file(
+        input_file=File(path=f"{s3_bucket}/homes.csv"),
+        output_table=Table(
+            conn_id="postgres_conn",
+        ),
+    )
+    sample_table = sample_create_table(my_homes_table)
+    df_res = my_df_func(sample_table)
+    aql.cleanup(tables_to_cleanup=[my_homes_table, sample_table, df_res])
+```
+Pros:
+* Requires no airflow knowledge
+* Users can control when the temp tables are deleted. They can delete it after the end of a task(s) or at the end of DAG.
+* Easy to debug, as tasks logs are accessible from the webserver as opposed to callbacks
+* Allows using tasks trigger rules which gives control of running cleanup even when certain task fails
+
+Cons:
+* If users add new tasks to the DAG, they need to remember to add it to the list
+* It might find all tables and delete them before they necessarily SHOULD be deleted if the downstream tasks are not added to the dependencies list
+
+### Option 4: a function with no parameters
+
+```python
+with dag:
+    my_homes_table = aql.load_file(
+        input_file=File(path=f"{s3_bucket}/homes.csv"),
+        output_table=Table(
+            conn_id="postgres_conn",
+        ),
+    )
+    sample_table = sample_create_table(my_homes_table)
+    df_res = my_df_func(sample_table)
+    [my_homes_table, sample_table, df_res] >> aql.cleanup()
+```
+
+Pros:
+* Would automatically find all tables
+* Easy to debug, as tasks logs are accessible from the webserver as opposed to callbacks
+* Allows using tasks trigger rules which gives control of running cleanup even when certain task fails
+
+Cons:
+* This ruins the feeling of the astro python SDK seeming “new”. We are exiting the functional paradigm and are now back in the land of bitshifting tasks.
+* If users add new tasks to the DAG, they need to remember to add it to the list
+* It might find all tables and delete them before they necessarily SHOULD be deleted if the downstream tasks are not added to the dependencies list
+
+### Option 5: A listener
+
+We looked into using the new airflow listener feature for this feature, but unfortunately found it was not compatible. Even if it was, it would require
```

Comment:
It may be worth adding a link and explaining why it is not compatible (ATM, in Airflow 2.3, listeners are at a task-level, and not at a DAG level)